File: backend/internal/domain/category.go
package domain

import "time"

type Category struct {
	ID           uint
	Name         string
	Description  string
	DisplayOrder int
	Active       bool
	CompanyID    uint // ID da empresa/tenant (obrigatório - Sprint 3)
	DeletedAt    *time.Time
	CreatedAt    time.Time
	UpdatedAt    time.Time
}

File: frontend/src/lib/types/category.ts
export interface Category {
  ID: number;
  Name: string;
  Description?: string;
  DisplayOrder: number;
  Active: boolean;
  CreatedAt?: string;
  UpdatedAt?: string;
}

export interface CategoryCreatePayload {
  name: string;
  description?: string;
  display_order: number;
}

export interface CategoryUpdatePayload {
  name: string;
  description?: string;
  display_order: number;
  active: boolean;
}
